### PR TITLE
Expose the methods required to implement a CA.F on base class.

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

**Comment by JakeWharton** [comment]
The `CompleteableFuture` adapter is done and depends on this one.

File: retrofit/src/main/java/retrofit2/CallAdapter.java [hunk 1]
```diff
@@ -57,11 +58,28 @@
...
      * Returns a call adapter for interface methods that return {@code returnType}, or null if it
      * cannot be handled by this factory.
      */
-    CallAdapter<?> get(Type returnType, Annotation[] annotations, Retrofit retrofit);
+    public abstract CallAdapter<?> get(Type returnType, Annotation[] annotations,
+        Retrofit retrofit);
+
+    /**
+     * Extract the upper bound of the generic parameter at {@code index} from {@code type}. For
+     * example, index 1 of {@code Map<String ? extends Runnable>} returns {@code Runnable}.
```
Comment (swankjesse) [inline retrofit/src/main/java/retrofit2/CallAdapter.java]: Missing comma?

**Comment by swankjesse** [comment]
LGTM

File: retrofit/src/main/java/retrofit2/CallAdapter.java (hunk 1 above)
Comment (JakeWharton) [inline retrofit/src/main/java/retrofit2/CallAdapter.java]: Laser vision!

**Pull request merged**